syntax = "proto3";

package stonkinator;

import "general_messages.proto";
import "securities_service.proto";

option go_package = "./stonkinator_rpc_service";

service DataFrameService {
    rpc MapTradingSystemInstrument (OperateOn) returns (CUD);
    rpc PushPrice (Price) returns (CUD);
    rpc PushPriceStream (stream Price) returns (CUD);
    rpc SetMinimumRows (MinimumRows) returns (CUD);
    rpc CheckPresence (GetBy) returns (Presence);
    rpc Evict (OperateOn) returns (CUD);
}

message MinimumRows {
    OperateOn operate_on  = 1;
    uint32 num_rows = 2;
}

message Presence {
    bool is_present = 1;
}